cmd/sriov/main.go

```diff
@@ -102,12 +101,27 @@ func cmdAdd(args *skel.CmdArgs) error {
 	}}
 
 	if !netConf.DPDKMode {
-		macAddr, err = sm.SetupVF(netConf, args.IfName, netns)
+		err = sm.SetupVF(netConf, args.IfName, netns)
 
 		if err != nil {
 			return fmt.Errorf("failed to set up pod interface %q from the device %q: %v", args.IfName, netConf.Master, err)
 		}
-		result.Interfaces[0].Mac = macAddr
+	}
+
+	// add the mac address we report to the CNI call return object
+	// if the device is on kernel mode we report that one back
+	// if not we check the administrative mac address on the PF
+	// if it is set and is not zero, report it.
+	if netConf.MAC != "" {
```

Comment:
can you refactor these nested ifs to a function to increase readability?

something like:

netConf.MAC = someFunc(netConf)

func someFunc(netConf) string {
  if netConf.MAC != "" {
    return netConf.MAC
  }

  if !netConf.DPDKMode {
    return netConf.OrigVfState.EffectiveMAC
  }

  if netConf.OrigVfState.AdminMAC != "00:00:00:00:00:00" {
    return netConf.OrigVfState.AdminMAC
  }
}
```

Reply:
+1

Reply:
I like the idea I also manage to unit test that way :)

Reply:
Great! Thanks